cornac/serving/app.py

```diff
@@ -151,5 +154,83 @@ def add_feedback():
     return jsonify(data), 200
 
 
+# curl -X POST -H "Content-Type: application/json" -d '{"metrics": ["RMSE()", "NDCG(k=10)"]}' "http://localhost:8080/evaluate"
+@app.route("/evaluate", methods=["POST"])
+def evaluate():
+    global model, train_set
+
+    if model is None:
+        return "Model is not yet loaded. Please try again later.", 400
+
+    if train_set is None:
+        return "Unable to evaluate. 'train_set' is not provided", 400
+
+    query = request.json
+
+    query_metrics = query.get("metrics")
+    rating_threshold = query.get("rating_threshold", 1.0)
+    exclude_unknowns = (
+        query.get("exclude_unknowns", "true").lower() == "true"
+    )  # exclude unknown users/items by default, otherwise specified
+    user_based = (
+        query.get("user_based", "true").lower() == "true"
+    )  # user_based evaluation by default, otherwise specified
+
+    if query_metrics is None:
+        return "metrics is required", 400
+    elif not isinstance(query_metrics, list):
+        return "metrics must be an array of metrics", 400
+
+    # organize metrics
+    metrics = []
+    for i, metric in enumerate(query_metrics):
+        try:
+            metrics.append(eval(metric))
+        except:
+            return f"Invalid metric initiation: {metric}", 400
```

Comment:
Maybe provide example of correct metric initiation/construction (e.g., 'RSME()', 'Recall(k=10)')?

Reply:
Although, it's very convenient to use `eval` function, there are concerns about security for this case since it'll be exposed as an API call. We probably want to change to a different way to parse the metric input.

Reply:
added `_safe_eval()` to mitigate security risks. More details: https://realpython.com/python-eval-function